GG/GG/Font.h

```diff
@@ -147,6 +151,13 @@ class GG_API Font
             pair.second. */
         Substring(const std::string& str_, const IterPair& pair);
 
+        /** Attach this Substring to \p str_.
+
+            This changes the iterators from pointing into the previous
```

Comment:
Yes, it is misleading.  

I do not think that the iterators are ever currently stored.  But they could be.

It could just say that previous iterators are invalidated and any use will have indeterminate effects, then any use after rebinding it the responsibility of the programmer.